open_spiel/algorithms/infostate_tree_test.cc

```diff
@@ -311,13 +309,60 @@ void CheckTreeLeaves(const InfostateTree& tree, int move_limit) {
   }
 }
 
+void CheckContinuation(const InfostateTree& tree) {
+  const std::vector<InfostateNode*>& leaves =
+      tree.nodes_at_depth(tree.tree_height());
+  std::shared_ptr<InfostateTree> continuation = MakeInfostateTree(leaves);
+
+  SPIEL_CHECK_EQ(continuation->root_branching_factor(), leaves.size());
+  for (int i = 0; i < leaves.size(); ++i) {
+    const InfostateNode* leaf_node = leaves[i];
+    const InfostateNode* root_node = continuation->root().child_at(i);
+    SPIEL_CHECK_TRUE(leaf_node->is_leaf_node());
+    if (leaf_node->type() != kTerminalInfostateNode) {
+      SPIEL_CHECK_EQ(leaf_node->type(), root_node->type());
+      SPIEL_CHECK_EQ(leaf_node->has_infostate_string(),
+                     root_node->has_infostate_string());
+      if (leaf_node->has_infostate_string()) {
+        SPIEL_CHECK_EQ(leaf_node->infostate_string(),
+                       root_node->infostate_string());
+      }
+    } else {
+      // If the leaf node is terminal, the continuation might put this node
+      // deeper than in the root due to tree balancing with other leaf
+      // non-terminal nodes. Therefore we check whether (the possibly occurring)
+      // chain of dummy observations leads to this terminal node.
+      InfostateNode* terminal_continuation = continuation->root().child_at(i);
+      while (terminal_continuation->type() == kObservationInfostateNode) {
+        SPIEL_CHECK_FALSE(terminal_continuation->is_leaf_node());
+        SPIEL_CHECK_EQ(terminal_continuation->num_children(), 1);
+        terminal_continuation = terminal_continuation->child_at(0);
+      }
+      SPIEL_CHECK_EQ(terminal_continuation->type(), kTerminalInfostateNode);
+      SPIEL_CHECK_EQ(leaf_node->has_infostate_string(),
+                     terminal_continuation->has_infostate_string());
+      if (leaf_node->has_infostate_string()) {
+        SPIEL_CHECK_EQ(leaf_node->infostate_string(),
+                       terminal_continuation->infostate_string());
+      }
+      SPIEL_CHECK_EQ(leaf_node->terminal_utility(),
```

Comment:
The pattern on checking if `leaf_node` and `terminal_continuation` return the same value from some method call repeat here too many times. Consider if this isn't a situation where an overload of the equality operator wouldn't be better. In any case, you can make a lambda that would be called only as `checkEquality(&InfostateNode::terminal_utility);`.

Reply:
operator== is tricky to get right, so I wouldn't do it unless there is a strong use-case for it -- this test isn't, as it's not quite doing exactly ==